src/main/java/org/b612foundation/adam/propagators/ClasspathStreamFactory.java

```diff
@@ -0,0 +1,39 @@
+package org.b612foundation.adam.propagators;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+
+import agi.foundation.compatibility.MemoryStream;
+import agi.foundation.infrastructure.StreamFactory;
+
+public class ClasspathStreamFactory extends StreamFactory {
```

Comment:
May not be needed here at all, or at least should get bundled with other STK stuff, possibly into a subpackage.